code/modules/mining/lavaland/megafauna_loot.dm

```diff
@@ -1049,6 +1049,13 @@
 		for(var/mob/living/hit_mob in turf)
 			to_chat(hit_mob, span_userdanger("You've been struck by lightning!"))
 			hit_mob.electrocute_act(15 * (isanimal(hit_mob) ? 3 : 1) * (turf == target ? 2 : 1) * (boosted ? 2 : 1), src, flags = SHOCK_TESLA|SHOCK_NOSTUN)
+			if(!ishostile(hit_mob))
+				continue
+
+			var/mob/living/simple_animal/hostile/hit_hostile = hit_mob
+			if(hit_hostile.stat == CONSCIOUS && !hit_hostile.target && hit_hostile.AIStatus != AI_OFF && !hit_hostile.client) //Mimicks what would happen on a melee attack
```

Comment:
I don't really like how this is snowflaked, maybe instead this could be on hostile's electrocute act, with a living check for source? and swap the electrocute_act above to be user instead of src.

Reply:
yeah you can't do that, source is supposed to be the actual source